src/plugins/intel_cpu/src/emitters/snippets/x64/jit_parallel_loop_emitters.cpp

```diff
@@ -0,0 +1,417 @@
+    if (ov::snippets::utils::is_dynamic_value(loop_args.m_work_amount)) {
+        jit_aux_gpr_holder gpr_holder(h, aux_gpr_idxs, out);
+        Reg64 reg_loop_args_ptr = gpr_holder.get_reg();
+        h->mov(reg_loop_args_ptr, h->ptr[abi_param1 + GET_OFF(loop_args)]);
+        h->mov(reg_work_amount, h->ptr[reg_loop_args_ptr + loop_id_offset + GET_OFF_LOOP_ARGS(m_work_amount)]);
+    } else {
+        h->mov(reg_work_amount, loop_args.m_work_amount);
+    }
+    // if wa < increment, skip the loop
+    // Note : If the loop should be evaluated once and increment is dynamic,
+    //        we should manually set `increment = 1` to compare the dynamic work amount
+    //        with `1` at least before loop execution
+    //        (work amount can be zero and we should skip this loop even `evaluate_once = 1`)
+    auto increment = evaluate_once && snippets::utils::is_dynamic_value(wa_increment) ? 1 : wa_increment;
+    h->cmp(reg_work_amount, increment);
+    h->jl(*loop_end_label, Xbyak::CodeGenerator::T_NEAR);
+
+    std::vector<Xbyak::Reg> regs_to_restore;
+    emit_parallel_executor_call(regs_to_restore);
+    // Note: parallel region starts here. The only legal entry point is from ParallelLoopExecutor::execute(...)
+    emit_parallel_region_initialization(regs_to_restore);
+}
+
+jit_parallel_loop_end_emitter::jit_parallel_loop_end_emitter(jit_generator_t* h,
+                                                             cpu_isa_t isa,
+                                                             const ov::snippets::lowered::ExpressionPtr& expr)
+    : jit_parallel_loop_base_emitter(h, isa, expr),
+      loop_begin_label{nullptr},
+      loop_end_label{new Label()} {
+    in_out_type_ = emitter_in_out_map::gpr_to_gpr;
+    const auto loop_end = ov::as_type_ptr<snippets::op::ParallelLoopEnd>(expr->get_node());
+    OV_CPU_JIT_EMITTER_ASSERT(loop_end != nullptr, "expected LoopEnd expr");
+    const auto begin_expr = get_loop_begin_expr(expr);
+    const auto& loop_begin_emitter =
+        std::dynamic_pointer_cast<jit_parallel_loop_begin_emitter>(begin_expr->get_emitter());
+    OV_CPU_JIT_EMITTER_ASSERT(loop_begin_emitter, "LoopBegin expected jit_loop_begin_emitter");
+    loop_begin_emitter->set_loop_end_label(loop_end_label);
+    loop_begin_label = loop_begin_emitter->get_begin_label();
+    m_parallel_section_reg_spiller = loop_begin_emitter->get_parallel_section_reg_spiller();
+}
+
+ov::snippets::lowered::ExpressionPtr jit_parallel_loop_end_emitter::get_loop_begin_expr(
+    const ov::snippets::lowered::ExpressionPtr& expr) {
+    auto begin_expr = expr->get_input_port_connectors().back()->get_source().get_expr();
+    OV_CPU_JIT_EMITTER_ASSERT(ov::is_type<snippets::op::ParallelLoopBegin>(begin_expr->get_node()),
+                              "LoopEnd expression must have th last port connector to LoopBegin");
+    return begin_expr;
+}
+
+void jit_parallel_loop_end_emitter::validate_arguments(const std::vector<size_t>& in,
```

Comment:
could you please add check for `m_parallel_section_reg_spiller != nullptr`?

Reply:
Thanks for the catch, the check is added